src/core/sys/posix/ucontext.d

```diff
@@ -921,19 +921,125 @@ else version (CRuntime_Musl)
 }
 else version (Darwin)
 {
-    alias mcontext_t = void;
-
-    struct ucontext
+    private
     {
-        int         uc_onstack;
-        sigset_t    uc_sigmask;
-        stack_t     uc_stack;
-        ucontext*   uc_link;
-        size_t      uc_mcsize;
-        mcontext_t* uc_mcontext;
+        version (X86_Any)
+        {
+            struct __darwin_mmst_reg
+            {
+                char[10] __mmst_reg;
+                char[6] __mmst_rsrv;
+            }
+
+            struct __darwin_xmm_reg
+            {
+                char[16] __xmm_reg;
+            }
+
+            struct __darwin_exception_state
+            {
+                uint __trapno;
+                uint __err;
+                version (X86_64)
+                    ulong __faultvaddr;
+                else
+                    uint __faultvaddr;
+            }
+
+            struct __darwin_thread_state
+            {
+                version (X86_64)
+                    ulong[21] __regs;
+                else
+                    uint[16] __regs;
+            }
+
+            struct __darwin_float_state
+            {
+                int[2] __fpu_reserved;
+                ushort __fpu_fcw;
+                ushort __fpu_fsw;
+                uint[7] __fpu_regs;
```

Comment:
It's difficult to review this when you're not following the headers.

Reply:
Check the sizeof(ucontext_t) from a C program.

Reply:
Yeah, I did. Again, I don't understand why you're doing it like this. Either follow the headers exactly or make the fields opaque.

Reply:
It would be 150 lines longer if I listed every single field.